okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java

```diff
@@ -104,6 +108,10 @@
 
   /** Nanotime timestamp when {@code allocations.size()} reached zero. */
   public long idleAtNanos = Long.MAX_VALUE;
+  private List<String> subjectAlternativeNames;
+
+  private Map<String, Boolean> supportedHostCache =
```

Comment:
I want to cache positive and negative results.  mainly because the previous lookup was exact match, mostly reference equality to find the right connection.  I wanted to avoid doing all the work on the 2nd, 3rd etc uses of a coalesced connection.